dds/DCPS/Marked_Default_Qos.cpp

```diff
@@ -19,9 +19,13 @@ int const INVALID_ENUM_VALUE = 999;
 DDS::DomainParticipantQos
 Marked_Default_Qos::marked_default_DomainParticipantQos()
 {
-  DDS::DomainParticipantQos qos;
-  DDS::DomainParticipantFactory_var factory = TheParticipantFactory;
-  factory->get_default_participant_qos(qos);
+  DDS::DomainParticipantQos qos = TheServiceParticipant->initial_DomainParticipantQos();
+  qos.user_data.value.length(5);
```

Comment:
couldn't get that working, that is why I went for a magic value, any suggestions to try?